mollie-payments-for-woocommerce/includes/mollie/wc/gateway/abstractsubscription.php

```diff
@@ -0,0 +1,594 @@
+
+            do_action(Mollie_WC_Plugin::PLUGIN_ID . '_create_payment', $data, $renewal_order);
+            $payment = null;
+            // Create Mollie payment with customer id.
+            try
+            {
+                Mollie_WC_Plugin::debug($this->id . ': Fetch mandate' . $renewal_order->id, true);
+                $mandates =  Mollie_WC_Plugin::getApiHelper()->getApiClient($test_mode)->customers_mandates->withParentId($customer_id)->all();
+                $validMandate = false;
+                foreach ($mandates as $mandate){
+                    if ($mandate->status == 'valid'){
+                        $validMandate = true;
+                        $data['method'] = $mandate->method;
+                        break;
+                    }
+                }
+                if ($validMandate){
+                    Mollie_WC_Plugin::debug($this->id . ': Valid mandate ' . $renewal_order->id, true);
+                    $payment = Mollie_WC_Plugin::getApiHelper()->getApiClient($test_mode)->payments->create($data);
+                } else {
+                    Mollie_WC_Plugin::debug($this->id . 'Payment problem ' . $renewal_order->id, true);
+                    throw new Mollie_API_Exception(__('Payment cannot be processed.', 'mollie-payments-for-woocommerce-mandate-problem'));
+                }
+            }
+            catch (Mollie_API_Exception $e)
+            {
+                if ($e->getField() !== 'customerId')
+                {
+                    throw $e;
+                }
+
+                // Retry without customer id.
+                unset($data['customerId']);
+                Mollie_WC_Plugin::debug($this->id . ': New debug ' . $renewal_order->id, true);
+                $payment = Mollie_WC_Plugin::getApiHelper()->getApiClient($test_mode)->payments->create($data);
+            }
+            Mollie_WC_Plugin::debug($this->id . ': Created payment for order ' . $renewal_order->id. ' payment json response '.json_encode($payment), true);
+            Mollie_WC_Plugin::getDataHelper()->unsetActiveMolliePayment($renewal_order->id);
+            // Set active Mollie payment
+            Mollie_WC_Plugin::getDataHelper()->setActiveMolliePayment($renewal_order->id, $payment);
+
+            // Set Mollie customer
+            $this->setUserMollieCustomerId($renewal_order->id, $payment->customerId);
+
+            do_action(Mollie_WC_Plugin::PLUGIN_ID . '_payment_created', $payment, $renewal_order);
+
+            Mollie_WC_Plugin::debug($this->id . ': Payment ' . $payment->id . ' (' . $payment->mode . ') created for order ' . $renewal_order->id);
+
+            // Set initial status
+            // Status is only updated if the new status is not the same as the default order status (pending)
+            $this->_updateScheduledPaymentOrder($renewal_order, $initial_order_status, $payment);
+
+
+            return array(+
```

Comment:
Excuses, dit is geen syntax error. De key wordt naar een integer gecast. Maar lijkt me sowieso niet de bedoeling.